### ISSUE TITLE
Bad memory allocation in develop branch

### ISSUE THREAD

**Issue by bemilio** [casadi/casadi#2495]
Following https://github.com/casadi/casadi/issues/2493 I installed the `develop` branch

I use the MATLAB bindings in Ubuntu 18.04
Running the following: 

```
import casadi.*

opti = Opti();

x = opti.variable(1);

cost = (x+5)^2;

opti.set_initial(x, -1);
opti.minimize(cost);
opti.solver('ipopt');
sol = opti.solve;
```

results in: 

> Error using casadiMEX
> Error in Opti::solve [OptiNode] at .../casadi/core/optistack.cpp:159:
> std::bad_alloc
> 
> Error in casadi.Opti/solve (line 181)
>       [varargout{1:nargout}] = casadiMEX(1242, self, varargin{:});
> 
> Error in SwigRef/subsref (line 26)
>             [varargout{1}] = builtin('subsref',self,substruct('.',s.subs,'()',{}));
> 
> Error in test_casadi (line 12)
> sol = opti.solve;

The same code works when using the `master` branch.

**Comment by jgillis** [comment]
Thanks for reporting! 
Does this occur with a freshly started Matlab, and `path` that contains only the develop CasADi?

How did you obtain a develop build?
What happens with a build from:
https://github.com/casadi/binaries/releases/tag/commit-450eba6

**Comment by bemilio** [comment]
Hi, thanks for the prompt answer. It looks like re-installing solved the issue, sorry for bothering.

The `Opti('conic')` works also well with `qpoases`

To answer your question and report you another thing, I followed the instructions at [https://github.com/casadi/casadi/wiki/matlab](this) link. I found that, during installation, `casadi` calls `/usr/local/swig3.0`which results in an installation error. I had to specify the installation path of `swig` and modify the swig executable path in the `casadi` CMake, in order to call an executable `path/swig` (without `3.0`). This solved the issue.
